virttest/qemu_monitor.py

```diff
@@ -2597,23 +2718,42 @@ def set_migrate_capability(self, state, capability):
         # clear if it's another reason for the error
         try:
             return self.cmd(cmd, args)
-        except QMPCmdError as e:
-            if e.data['class'] in ['GenericError']:
-                logging.debug(
-                    "Error in set_migrate_capability for %s: %s" % (capability, e))
-                raise MonitorNotSupportedMigCapError("set capability failed for %s (%s)" %
-                                                     (capability, e))
-            else:
-                raise e
-
-    def get_migrate_capability(self, capability):
+        except QMPCmdError as exc:
+            if disable_auto_x_evaluation:
+                raise
+            # Try it again with/without "x-" prefix
+            capability2 = x_non_x_feature(capability)
+            args = {"capabilities": [{"state": state,
+                                      "capability": capability2}]}
+            try:
+                return self.cmd(cmd, args)
+            except QMPCmdError as exc2:
+                logging.debug("Error in set_migrate_capability for %s: %s",
+                              capability, exc)
```

Comment:
should this be moved to L2723 ?

Reply:
Nop, at this point both attempts already failed so we log both failures and then postprocess the kind of failures and react accordingly.